### PR TITLE
Added blog on how to build new openstack packages

### PR DESCRIPTION
**Pull request by chkumar246**

### CODE REVIEW

**Comment by chkumar246** [comment]
@rbowen @apevec @hguemar Please have a look at the blog post so that we can publish it.

File: source/blog/2016-07-15-how-to-build-new-openstack-packages.html.md [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+
+[5.] Install DLRN
+
+    $ python setup.py develop
+
+[6.] Add the user you intend to run as to the mock group:
+
+    $ sudo usermod -a -G mock $USER
+    $ newgrp mock
+    $ newgrp $USER
```
Comment (hguemar) [inline source/blog/2016-07-15-how-to-build-new-openstack-packages.html.md]: unneeded, mock package does it for you.
http://pkgs.fedoraproject.org/cgit/rpms/mock.git/tree/mock.spec#n153

File: source/blog/2016-07-15-how-to-build-new-openstack-packages.html.md [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+Now your system is ready to use DLRN.
+
+## Let us package python foobar project for RDO
+
+[1.] create a project "foobar-spec" and initialize the project using git init
+
+    $ mkdir foobar-spec; cd foobar-spec
```
Comment (hguemar) [inline source/blog/2016-07-15-how-to-build-new-openstack-packages.html.md]: s/spec/distgit/ or no suffix

File: source/blog/2016-07-15-how-to-build-new-openstack-packages.html.md [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+[1.] go to .rdoinfo directory
+
+    $ cd ~/.rdopkg/rdoinfo
+
+[2.] Edit the rdoinfo.yml file and add package entry in the last
+
+    $ vim rdoinfo.yml
+    - project: foobar # project name
+      name: python-foobar # RDO package name
+      upstream: git://github.com/openstack/%(project)s # foobar project source code git repository
```
Comment (hguemar) [inline source/blog/2016-07-15-how-to-build-new-openstack-packages.html.md]: Either take an openstack project or use a placeholder url not to confuse readers

**Comment by chkumar246** [comment]
@hguemar I have updated the pull request, please have a look.

File: source/blog/2016-07-15-how-to-build-new-openstack-packages.html.md [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+
+    $ python setup.py develop
+
+Now your system is ready to use DLRN.
+
+## Let us package "congress" OpenStack project for RDO
+
+[1.] create a project "congress-distgit" and initialize the project using git init
+
+    $ mkdir congress-distgit; cd congress-distgit
```
Comment (yac) [inline source/blog/2016-07-15-how-to-build-new-openstack-packages.html.md]: I'd give `cd congress-distgit` its own line for the sake of clarity.

File: source/blog/2016-07-15-how-to-build-new-openstack-packages.html.md [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+    $ git init
+
+[2.] create a branch "rpm-master"
+
+    $ git checkout -b rpm-master
+
+[3.] Create the openstack-congress.spec file under congress-distgit directory by using RDO spec template : https://github.com/openstack-packages/openstack-example-spec and add it the spec file and commit it "rpm-master" branch.
```
Comment (yac) [inline source/blog/2016-07-15-how-to-build-new-openstack-packages.html.md]: Create `openstack-congress.spec` file using [RDO spec template](https://github.com/openstack-packages/openstack-example-spec) and commit it into `rpm-master` branch.

**Comment by hguemar** [comment]
:+1:

File: source/blog/2016-07-15-how-to-build-new-openstack-packages.html.md [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+
+    $ git add openstack-congress.spec
+    $ git commit -m "<your commit message>"
+
+## Add a package entry in rdoinfo
+
+[1.] go to .rdoinfo directory
+
+    $ cd ~/.rdopkg/rdoinfo
```
Comment (yac) [inline source/blog/2016-07-15-how-to-build-new-openstack-packages.html.md]: 1) you need to have used `rdopkg info` or related command before to have this at all
2) `rdopkg` will overwrite this without warning

Proper way would be to `rdopkg info && cp -r ~/.rdopkg/rdoinfo $SOMEWHERE_LOCAL`.

File: source/blog/2016-07-15-how-to-build-new-openstack-packages.html.md [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+            --package-name <openstack-congress> \ # --package to build
+            --head-only \ # To build the package using latest commit
+            --dev \ # to build the package locally
+
+[4.] Once build is completed, you can find the rpms and srpms in this folder
+
+    $ # path to packaged rpms and srpms
+    $ <path to DLRN>/data/repos/current/
+
+Now you can use the latest rpms and use it for testing.
```
Comment (yac) [inline source/blog/2016-07-15-how-to-build-new-openstack-packages.html.md]: use use... weird sentence :)

File: source/blog/2016-07-15-how-to-build-new-openstack-packages.html.md [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+
+## Run DLRN to build openstack-congress package
+
+[1.] Go to DLRN project directory.
+
+[2.] Run the following command to build the package
+
+    $ dlrn --config-file projects.ini \
+            --info-repo ~/.rdopkg/rdoinfo \ # --info-repo flag is used to point local rdoinfo repo
+            --package-name <openstack-congress> \ --package to build
```
Comment (yac) [inline source/blog/2016-07-15-how-to-build-new-openstack-packages.html.md]: This should be either `openstack-congress` or `<package_name>` as referenced bellow. I'd go with `openstack-congress` here and...

File: source/blog/2016-07-15-how-to-build-new-openstack-packages.html.md [hunk 9]
```diff
@@ -0,0 +1,109 @@
...
+            --head-only \ To build the package using latest commit
+
+It will clone the project code <package_name> and spec under <package_name>_distro folder.
```
Comment (yac) [inline source/blog/2016-07-15-how-to-build-new-openstack-packages.html.md]: ...here.

**Comment by rbowen** [comment]
This looks good to me. The only comment that I'd make is to encourage you to link to some other pages, including the DLRN full documentation, and possibly further information about RDO itself. Remember that people will read this blog post, without the site context, on planet.openstack.org and other places.

**Pull request merged**